libs/remix-ui/editor/src/lib/providers/hoverProvider.ts

```diff
@@ -145,6 +145,48 @@ export class RemixHoverProvider implements languages.HoverProvider {
             getLinks(nodeAtPosition)
             getDocs(nodeAtPosition)
             // getScope(nodeAtPosition)
+
+            try {
+                if (nodeAtPosition?.name === 'msg') {
+                    const global = await this.props.plugin.call('debugger', 'globalContext')
+                    if (global !== null && global[nodeAtPosition?.name]) {
+                        contents.push({
+                            value: `${nodeAtPosition.name} = ${JSON.stringify(global[nodeAtPosition?.name], null, '\t')}`
+                        })
+                    }
+                }
+            } catch (e) {}
+
+            try {
+                if (nodeAtPosition?.expression?.name === 'msg' && nodeAtPosition?.memberName) {
+                    const global = await this.props.plugin.call('debugger', 'globalContext')
+                    if (global !== null && global[nodeAtPosition?.expression?.name][nodeAtPosition.memberName] && global[nodeAtPosition?.expression?.name][nodeAtPosition.memberName]) {
+                        contents.push({
+                            value: `${nodeAtPosition.memberName} = ${global[nodeAtPosition?.expression?.name][nodeAtPosition.memberName]}`
+                        })
+                    }
+                }
+            } catch (e) {}
+
+            try {
+                const decodedVar = await this.props.plugin.call('debugger', 'decodeLocalVariable', nodeAtPosition.id)
```

Comment:
the nodes parser knows about might have different ids as the parser compiler runs continuously when typing

Reply:
so this would cause a bug